src/mpi/romio/adio/ad_daos/ad_daos_io_str.c

```diff
@@ -0,0 +1,434 @@
+        fprintf(stderr, "NUM IO lists = %d\n", n_write_lists);
+#endif
+
+        for (i = 0; i < n_write_lists; i++) {
+            if (!i) {
+                rgs[i].rg_idx = start_off;
+                rgs[i].rg_len = MPL_MIN(f_data_wrote, st_fwr_size);
+                if (request)
+                    aio_req->nbytes += rgs[i].rg_len;
+#ifdef D_PRINT_IO
+                printf("(%d) %d: idx %lld len %zu\n", mpi_rank, i, rgs[i].rg_idx, rgs[i].rg_len);
+#endif
+            } else {
+                if (flat_file->blocklens[j]) {
+                    rgs[i].rg_idx = disp +
+                        ((ADIO_Offset) n_filetypes) * filetype_extent + flat_file->indices[j];
+                    rgs[i].rg_len = flat_file->blocklens[j];
+#ifdef D_PRINT_IO
+                    printf("(%d) %d: idx %lld len %zu\n",
+                           mpi_rank, i, rgs[i].rg_idx, rgs[i].rg_len);
+#endif
+                    if (request)
+                        aio_req->nbytes += rgs[i].rg_len;
+                } else
+                    i--;
+            }
+
+            if (j < (flat_file->count - 1))
+                j++;
+            else {
+                j = 0;
+                n_filetypes++;
+            }
+        }
+    }
+
+    /** set array location */
+    iod->arr_nr = n_write_lists;
+    iod->arr_rgs = rgs;
+    if (request)
+        aio_req->rgs = rgs;
+
+    if (rw_type == DAOS_WRITE) {
+        ret = daos_array_write(cont->oh, DAOS_TX_NONE, iod, sgl, NULL,
+                               (request ? &aio_req->daos_event : NULL));
+        if (ret != 0) {
+            PRINT_MSG(stderr, "daos_array_write() failed with %d\n", ret);
+            *error_code = ADIOI_DAOS_err(myname, cont->obj_name, __LINE__, ret);
+            return;
+        }
+    } else if (rw_type == DAOS_READ) {
+        ret = daos_array_read(cont->oh, DAOS_TX_NONE, iod, sgl, NULL,
+                              (request ? &aio_req->daos_event : NULL));
+        if (ret != 0) {
+            PRINT_MSG(stderr, "daos_array_read() failed with %d\n", ret);
+            *error_code = ADIOI_DAOS_err(myname, cont->obj_name, __LINE__, ret);
+            return;
+        }
+    }
+
+    if (file_ptr_type == ADIO_INDIVIDUAL) {
+        if (filetype_is_contig)
+            fd->fp_ind += bufsize;
+        else
+            fd->fp_ind = rgs[n_write_lists - 1].rg_idx + rgs[n_write_lists - 1].rg_len;
+    }
+
+    if (!err_flag)
+        *error_code = MPI_SUCCESS;
+
+    fd->fp_sys_posn = -1;       /* clear this. */
+
+#ifdef HAVE_STATUS_SET_BYTES
+    if (status)
+        MPIR_Status_set_bytes(status, datatype, bufsize);
+#endif
+
+    if (!request) {
+        ADIOI_Free(iovs);
+        ADIOI_Free(rgs);
+    }
+
+    return;
+}
+
+void ADIOI_DAOS_ReadStridedColl(ADIO_File fd, void *buf, int count,
+                                MPI_Datatype datatype, int file_ptr_type,
+                                ADIO_Offset offset, ADIO_Status * status, int *error_code)
+{
+    MPI_Barrier(fd->comm);
```

Comment:
Huh.  The only reason you implement ADIOI_DAOS_ReadStridedColl instead of calling ADIOI_GEN_ReadStridedColl is so you can barrier before the general collective is called?   If this was left behind as a debugging aide, I can delete it before merging.  If it's needed, please add comment  why.

Reply:
yes those are not required, let me push an update to the PR to remove those.